port/common/omrsysinfo_helpers.c

```diff
@@ -218,10 +206,136 @@ omrsysinfo_get_x86_description(struct OMRPortLibrary *portLibrary, OMRProcessorD
 	/* features */
 	desc->features[0] = CPUInfo[CPUID_EDX];
 	desc->features[1] = CPUInfo[CPUID_ECX];
+	desc->features[2] = 0; /* reserved for future expansion */
+	/* extended features */
+	omrsysinfo_get_x86_cpuid_ext(CPUID_STRUCTURED_EXTENDED_FEATURE_INFO, 0, CPUInfo); /* 0x0 is the only valid subleaf value for this leaf */
+	desc->features[3] = CPUInfo[CPUID_EBX]; /* Structured Extended Feature Flags in EBX */
 
-	cpuidex(CPUInfo, 7, 0);
-	desc->features[2] = CPUInfo[CPUID_EBX];
-
+	desc->features[4] = 0; /* reserved for future expansion */
 	return 0;
 }
+
+/**
+ * Assembly code to get the register data from CPUID instruction
+ * This function executes the CPUID instruction based on which we can detect
+ * if the environment is virtualized or not, and also get the Hypervisor Vendor
+ * Name based on the same instruction. The leaf value specifies what information
+ * to return.
+ *
+ * @param[in]   leaf        The leaf value to the CPUID instruction and the value
+ *                          in EAX when CPUID is called. A value of 0x1 returns basic
+ *                          information including feature support.
+ * @param[out]  cpuInfo     Reference to the an integer array which holds the data
+ *                          of EAX,EBX,ECX and EDX registers.
+ *              cpuInfo[0]  To hold the EAX register data, value in this register at
+ *                          the time of CPUID tells what information to return
+ *                          EAX=0x1,returns the processor Info and feature bits
+ *                          in EBX,ECX,EDX registers.
+ *                          EAX=0x40000000 returns the Hypervisor Vendor Names
+ *                          in the EBX,ECX,EDX registers.
+ *              cpuInfo[1]  For EAX = 0x40000000 hold first 4 characters of the
+ *                          Hypervisor Vendor String
+ *              cpuInfo[2]  For EAX = 0x1, the 31st bit of ECX tells if its
+ *                          running on Hypervisor or not,For EAX = 0x40000000 holds the second
+ *                          4 characters of the the Hypervisor Vendor String
+ *              cpuInfo[3]  For EAX = 0x40000000 hold the last 4 characters of the
+ *                          Hypervisor Vendor String
+ *
+ */
+void
+omrsysinfo_get_x86_cpuid(uint32_t leaf, uint32_t *cpuInfo)
+{
+	cpuInfo[0] = leaf;
+
+#if defined(OMR_OS_WINDOWS)
+	/* Specific CPUID instruction available in Windows */
+	__cpuid(cpuInfo, cpuInfo[0]);
+
+#elif defined(LINUX) || defined(OSX)
+#if defined(J9X86)
+	__asm volatile(
+		"mov %%ebx, %%edi;"
+		"cpuid;"
+		"mov %%ebx, %%esi;"
+		"mov %%edi, %%ebx;"
+		:"+a" (cpuInfo[0]), "=S" (cpuInfo[1]), "=c" (cpuInfo[2]), "=d" (cpuInfo[3])
+		: /* None */
+		:"edi");
+
+#elif defined(J9HAMMER)
+	__asm volatile(
+     	"cpuid;"
+     	:"+a" (cpuInfo[0]), "=b" (cpuInfo[1]), "=c" (cpuInfo[2]), "=d" (cpuInfo[3])
+        );
```

Comment:
This one is still indented with spaces, should be tabs. And the `#endif`s below should be commented with the condition.